pennylane/transforms/defer_measurements.py

```diff
@@ -23,7 +23,7 @@
 from pennylane.wires import Wires
 from pennylane.queuing import QueuingManager
 
-# pylint: disable=too-many-branches
+# pylint: disable=too-many-branches, too-many-statements
```

Comment:
Out of scope for this PR, but we might want to consider starting to breaking out helper functions.

Reply:
I have a follow up PR where I'm going to break it down. Haven't done it yet but it's part of what I want to improve in `defer_measurements`. See [here](https://github.com/PennyLaneAI/pennylane/pull/4625)